src/Content/OEmbed.php

```diff
@@ -49,132 +48,128 @@ public static function replaceCallback($matches)
-
-			if (!$txt || $txt[0] != "{") {
-				$txt = '{"type":"error"}';
-			} else { //save in cache
-				$j = json_decode($txt);
-				if (!empty($j->type) && $j->type != "error") {
-					DBA::insert('oembed', [
-						'url' => normalise_link($embedurl),
-						'maxwidth' => $a->videowidth,
-						'content' => $txt,
-						'created' => DateTimeFormat::utcNow()
-					], true);
-				}
+			$json_string = trim($json_string);
 
-				Cache::set($a->videowidth . $embedurl, $txt, CACHE_DAY);
+			if (!$json_string || $json_string[0] != '{') {
+				$json_string = '{"type":"error"}';
 			}
-		}
 
-		$j = json_decode($txt);
+			$oembed->parseJSON($json_string);
+			if (!empty($oembed->type) && $oembed->type != 'error') {
+				DBA::insert('oembed', [
+					'url' => normalise_link($embedurl),
+					'maxwidth' => $a->videowidth,
+					'content' => $json_string,
+					'created' => DateTimeFormat::utcNow()
+				], true);
+			}
 
-		if (!is_object($j)) {
-			return false;
+			Cache::set($cache_key, $json_string, CACHE_DAY);
 		}
 
-		// Always embed the SSL version
-		if (isset($j->html)) {
-			$j->html = str_replace(["http://www.youtube.com/", "http://player.vimeo.com/"], ["https://www.youtube.com/", "https://player.vimeo.com/"], $j->html);
+		if ($oembed->type == 'error') {
+			return $oembed;
 		}
 
-		$j->embedurl = $embedurl;
+		// Always embed the SSL version
+		$oembed->html = str_replace(['http://www.youtube.com/', 'http://player.vimeo.com/'], ['https://www.youtube.com/', 'https://player.vimeo.com/'], $oembed->html);
 
 		// If fetching information doesn't work, then improve via internal functions
-		if ($no_rich_type && ($j->type == "rich")) {
+		if ($no_rich_type && ($oembed->type == 'rich')) {
 			$data = ParseUrl::getSiteinfoCached($embedurl, true, false);
-			$j->type = $data["type"];
+			$oembed->type = $data['type'];
 
-			if ($j->type == "photo") {
-				$j->url = $data["url"];
+			if ($oembed->type == 'photo') {
+				$oembed->url = $data['url'];
 			}
 
-			if (isset($data["title"])) {
-				$j->title = $data["title"];
+			if (isset($data['title'])) {
+				$oembed->title = $data['title'];
 			}
 
-			if (isset($data["text"])) {
-				$j->description = $data["text"];
+			if (isset($data['text'])) {
+				$oembed->description = $data['text'];
 			}
 
-			if (is_array($data["images"])) {
-				$j->thumbnail_url = $data["images"][0]["src"];
-				$j->thumbnail_width = $data["images"][0]["width"];
-				$j->thumbnail_height = $data["images"][0]["height"];
+			if (is_array($data['images'])) {
+				$oembed->thumbnail_url = $data['images'][0]['src'];
+				$oembed->thumbnail_width = $data['images'][0]['width'];
+				$oembed->thumbnail_height = $data['images'][0]['height'];
 			}
 		}
 
-		Addon::callHooks('oembed_fetch_url', $embedurl, $j);
+		Addon::callHooks('oembed_fetch_url', $embedurl, $oembed);
 
-		return $j;
+		return $oembed;
 	}
 
-	private static function formatObject(stdClass $j)
+	private static function formatObject(\Friendica\Object\OEmbed $oembed)
```

Comment:
Please put a `use Friendica\Object\OEmbed` in head of this file.

Reply:
It conflicts with the class name it's used in.

Best I can do is either 
```php
use Friendica\Object;

function formatObject(Object\OEmbed $oembed) {
...
}
```
or
```php
use Friendica\Object\OEmbed as OEmbedObject;

function formatObject(OEmbedObject $oembed) {
...
}
```

Reply:
Then `OEmbed` is maybe not the right choice after all due to class name (without namespace) conflict. Maybe rename the name class to `OEmbedObject` then?

Reply:
That's the point of namespaces, you can have meaningful short class names organized in meaningful folders and still be able to use them together.